packages/hardhat-verify/src/internal/tasks/sourcify.ts

```diff
@@ -1,28 +1,179 @@
 import chalk from "chalk";
 import { subtask } from "hardhat/config";
+import { Sourcify } from "../sourcify";
+import { TASK_COMPILE } from "hardhat/builtin-tasks/task-names";
+
+import {
+  BuildInfoNotFoundError,
+  ContractNotFoundError,
+  ContractVerificationFailedError,
+  VerificationAPIUnexpectedMessageError,
+} from "../errors";
 
 import {
   TASK_VERIFY_SOURCIFY,
+  TASK_VERIFY_SOURCIFY_ATTEMPT_VERIFICATION,
   TASK_VERIFY_SOURCIFY_DISABLED_WARNING,
 } from "../task-names";
 
+interface VerificationResponse {
+  success: boolean;
+  message: string;
+}
+
+interface VerificationArgs {
+  address: string;
+  constructorArgs: string[];
+  contractFQN?: string;
+}
+
+interface VerificationInterfaceVerifyParams {
+  address: string;
+  files?: {
+    [index: string]: string;
+  };
+  chosenContract?: number;
+}
+
+interface VerificationInterface {
+  isVerified(address: string): Promise<Boolean>;
+  verify(params: VerificationInterfaceVerifyParams): Promise<any>;
+  getVerificationStatus(guid: string): Promise<any>;
+  getContractUrl(address: string, status?: string): string;
+}
+
+interface AttemptVerificationArgs {
+  address: string;
+  verificationInterface: VerificationInterface;
+  contractFQN: string;
+}
```

Comment:
The verification interface is Sourcify. Don't redefine it, you can do:
```suggestion

interface AttemptVerificationArgs {
  address: string;
  verificationInterface: Sourcify;
  contractFQN: string;
}
```